### PR TITLE
feat(eks): default masters role

### LINKED ISSUE #9463: [eks] Default masters role
Today, in order to issue `kubectl` commands against an EKS cluster, the `mastersRole` must be explicitly defined. This creates quite a lot of friction for users.

Moreoever, the `kubectlEnabled` option was confused to "solve" this problem because it falls back to use the stock CFN resource, which grants masters permissions to the user that _creates_ the cluster (so kubectl commands can be issued).

To solve this, we should simply auto-generate a masters role if a role is not defined.

### PR DESCRIPTION
**Pull request by eladb**
If `mastersRole` is not specified, we now define a default IAM role that can be assumed by anyone (with permissions) in the account.
        
This will allow users to interact with the cluster through `kubectl` by issuing the `aws eks update-kubeconfig` command with the appropriate `--role-arn` option, as specified in the CFN output.
    
Fixes #9463

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-eks/README.md [hunk 1]
```diff
@@ -45,6 +47,45 @@ cluster.addResource('mypod', {
 });
 ```
 
+In order to interact with your cluster through `kubectl`, you can use the `aws
+eks update-kubeconfig` [AWS CLI](https://aws.amazon.com/cli/) command in order
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-eks/README.md:51]:
```suggestion
eks update-kubeconfig` [AWS CLI](https://docs.aws.amazon.com/cli/latest/reference/eks/update-kubeconfig.html) command
```

File: packages/@aws-cdk/aws-eks/README.md [hunk 2]
```diff
@@ -45,6 +47,45 @@ cluster.addResource('mypod', {
...
+$ kubectl get all -n kube-system
+NAME                           READY   STATUS    RESTARTS   AGE
+pod/aws-node-fpmwv             1/1     Running   0          21m
+pod/aws-node-m9htf             1/1     Running   0          21m
+pod/coredns-5cb4fb54c7-q222j   1/1     Running   0          23m
+pod/coredns-5cb4fb54c7-v9nxx   1/1     Running   0          23m
+...
+```
+
+[RBAC]: https://kubernetes.io/docs/reference/access-authn-authz/rbac/
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-eks/README.md:87]: Why here? You can inline this inside the appropriate block quote

File: packages/@aws-cdk/aws-eks/lib/cluster.ts [hunk 3]
```diff
@@ -689,19 +686,25 @@ export class Cluster extends Resource implements ICluster {
...
-    // map the IAM role to the `system:masters` group.
-    if (props.mastersRole) {
-      if (!this.kubectlEnabled) {
+    if (!this.kubectlEnabled) {
+      if (props.mastersRole) {
         throw new Error('Cannot specify a "masters" role if kubectl is disabled');
       }
+    } else {
+      // if an explicit role is not configured, define a masters role that can
+      // be assumed by anyone in the account (with sts:AssumeRole permissions of
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-eks/lib/cluster.ts:695]: I would mention the `sts:AssumeRole` explanation in the README, and possibly even in the default value for `mastersRole`. It was a bit vague for me until I reached this point.

File: packages/@aws-cdk/aws-eks/README.md (hunk 2 above)
Comment (eladb) [reply packages/@aws-cdk/aws-eks/README.md]: Done

File: packages/@aws-cdk/aws-eks/lib/cluster.ts (hunk 3 above)
Comment (eladb) [reply packages/@aws-cdk/aws-eks/lib/cluster.ts:695]: Done

**Pull request merged**